### PR TITLE
[macOS] Add ImageButton to macOS

### LINKED ISSUE #4792: [macOS] Implement ImageButton on macOS
### Description
I am trying to port a working Xamarin Forms 3.4x project to macOS. The ImageButtons do not display.

### Steps to Reproduce
I have the following XAML code. I even tried to get a border to be displayed on the macOS version, just so I know that some of the ImageButton logic is working, even if the image itself was not being displayed. However, nothing displays on macOS, not even the border.

```
<ImageButton x:Name="btnFirstQuestion" Source="TriangleFastBackward.png"
                        IsEnabled="{Binding Path=IsFirstQuestion, Converter={StaticResource inverseBooleanConverter}}" 
                        Clicked="OnFirstQuestionClicked">
            <ImageButton.BorderWidth>
                    <OnPlatform x:TypeArguments="x:Double">
                            <On Platform="macOS" Value="1" />
                    </OnPlatform>
            </ImageButton.BorderWidth>
</ImageButton>
```

### Expected Behavior
The ImageButton should be displayed with the associated image and a 1px border. Even if the image has difficulty displaying, the 1px border should be displayed.

### Actual Behavior
Nothing is displayed.

### Basic Information
Xamarin Forms 3.40.1008975

- Version with issue:
- Last known good version: None
- IDE: Visual Studio Mac Community 7.71
- Affected Devices: MacBook Air Mid 2012, Mohave 10.14.1

### PR DESCRIPTION
**Pull request by krdmllr**
### Description of Change ###

Implements ImageButton on macOS.
Adds Button/BorderManager and adjusts internal views to be used across iOS/macOS internally.

### Issues Resolved ###  

- fixes #4792

### API Changes ### 

 None

### Platforms Affected ### 

-macOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Sample for the working `ImageButton` next to the iOS sample.

grafik
With border color set:
grafik

### Testing Procedure ###

- Testable in the control gallery

### PR Checklist ### 

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by krdmllr** [comment]
Hi @hartez @samhouts can I help somehow to get this merged :)?

File: Xamarin.Forms.Platform.iOS/Renderers/ImageButtonRenderer.cs [hunk 1]
```diff
@@ -21,14 +33,16 @@ public class ImageButtonRenderer : ViewRenderer<ImageButton, UIButton>, IImageVi
 		// but under iOS that suggestion won't work
 		readonly nfloat _minimumButtonHeight = 44; // Apple docs
 
+		public bool IsDisposed => throw new NotImplementedException();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/ImageButtonRenderer.cs]: This is wrong

File: Xamarin.Forms.Platform.iOS/Renderers/IImageVisualElementRenderer.cs [hunk 2]
```diff
@@ -5,13 +5,24 @@ namespace Xamarin.Forms.Platform.iOS
 #else
 using NativeImageView = AppKit.NSImageView;
 using NativeImage = AppKit.NSImage;
+using CoreAnimation;
+
 namespace Xamarin.Forms.Platform.MacOS
 #endif
 {
+#if __MACOS__
+	public interface IImageView
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/IImageVisualElementRenderer.cs]: Do we need to make this public ?

File: Xamarin.Forms.Platform.iOS/Renderers/IImageVisualElementRenderer.cs [hunk 3]
```diff
@@ -5,13 +5,24 @@ namespace Xamarin.Forms.Platform.iOS
...
+
+		CALayer Layer { get; set; } 
+	}
+#endif
+
 	public interface IImageVisualElementRenderer : IVisualNativeElementRenderer
 	{
 		void SetImage(NativeImage image);
 		bool IsDisposed { get; }
-		NativeImageView GetImage();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/IImageVisualElementRenderer.cs]: This will break existing users that could be using this interface no?

**Comment by samhouts** [comment]
@krdmllr Are you able to address the feedback above? Thanks!

**Comment by krdmllr** [comment]
Yes, already started :)

**Comment by krdmllr** [comment]
@samhouts @rmarinho  I unfortunately wont come around finishing this until mid february. Feel free to finish it yourself if it is urgent, but I will definitely get back to it once I have more time :)

**Comment by samhouts** [comment]
I'm going to close this one for now until we're ready to return to it. Thanks, @krdmllr !

**Comment by mhrastegary77** [comment]
hi @krdmllr , it's been so long since mid february 2020 :( don't you wanna come back and finish it yet?
we really love to see it in XF 5 :)

**Pull request closed without merging**